GrocifyApp.UI/src/components/modals/ProductModal.jsx

```diff
@@ -0,0 +1,119 @@
+
+    const [measures, setMeasures] = useState([]);
+    //const [sections, setSections] = useState([]);
+
+    const { makeRequest } = useApiRequest();
+
+    useEffect(() => {
+        if (productName !== "") {
+            setButtonDisabled(false);
+        } else {
+            setButtonDisabled(true);
+        }
+    }, [productName]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const pMeasures = await getProductMeasures();
+            //const pSections = await getProductSections();
+
+            if (pMeasures) {
+                setProductMeasure(pMeasures[0]);
+            }
+
+            //if (pSections) {
+            //    setProductSection(pSections[0]);
+            //}
+        };
+
+        fetchData();
+    }, []);
+
+    const getProductMeasures = async () => {
+        try {
+            const responseData = await makeRequest(ApiEndpoints.ProductMeasures_Endpoint, 'GET');
+            setMeasures(responseData);
+            return responseData;
+        } catch (error) {
+            console.log(error);
+        }
+    };
+
+    //const getProductSections = async () => {
+    //    try {
+    //        const responseData = await makeRequest(ApiEndpoints.ProductSections_Endpoint, 'GET');
+    //        setSections(responseData);
+    //        return responseData;
+    //    } catch (error) {
+    //        console.log(error);
+    //    }
+    //};
+
+    const createProduct = async () => {
+        const section = productSections.find(s => s.id === productSection.id);
+        const measure = measures.find(s => s.id === productMeasure.id);
+
+        const data = { name: productName, productSectionId: section.id, productSection: section, productMeasureId: measure.id, productMeasure: measure };
+
+        await makeRequest(ApiEndpoints.Products_Endpoint, 'POST', data);
+
+        onConfirm(data.productSectionId);
+    };
+
+    return (
+        <BaseModal isOpen={true} onClose={onClose} onConfirm={createProduct} isButtonDisabled={isButtonDisabled}
+            buttonText={ButtonConsts.Create} titleModal={ModalConsts.NewProduct} modalBody={
+                <div className={styles.inputRow}>
+                    <CustomInputApp className="app-form mb-0"
+                        type={InputType.Input}
+                        placeholder={PlaceholderConsts.AddProductName}
+                        label={LabelConsts.ProductName}
+                        value={productName}
+                        onChange={(e) => setProductName(e.target.value)}
+                        isRequired={true} />
+
+                    <ProductSectionSelector
+                        selectedValue={{ id: productSection.id, name: productSection.name, icon: productSection.icon }}
+                        selectedValueChanged={(e) => setProductSection(e)}
+                        productSections={productSections}
+                        isViewList={true} />
+
+                    <ProductMeasureSelector
```

Comment:
do something like this:

{productMeasure.name && (<ProductMeasureSelector
    selectedValue={{ id: productMeasure.id, name: productMeasure.name }}
    selectedValueChanged={(e) => setProductMeasure(e)}
    productMeasures={measures} />)}
    
Because productMeasures are being called using an async method, and this is rendered before getting the productMeasure, throwing the following exception:

![image](https://github.com/Catarina98/Grocify/assets/36730010/41f535a9-e4e7-4e8a-a113-5d53d7134b85)